src/videojs-hls.js

```diff
@@ -1205,7 +1288,7 @@ videojs.HlsHandler.prototype.updateEndHandler_ = function () {
     return;
   }
 
-  playlist = this.playlists.media();
+  playlist = segmentInfo.playlist;
```

Comment:
There are two reasons for this change:
1. It just makes it consistent that all the values we are pulling into local variables come from `segmentInfo`
1. I had planned to move `maybeEndOfStream_` functionality to *before* the early exit if we switched renditions. I wanted to signal the end-of-stream no matter what happens if we need to... 

Obviously, #2 never happened but since I shall be changing the function anyway given your other comment, I might as well do that too!